plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/catalog/hms/TestIcebergHiveMetastoreTableOperationsReleaseLockFailure.java

```diff
@@ -0,0 +1,130 @@
+import io.trino.plugin.iceberg.TestingIcebergConnectorFactory;
+import io.trino.spi.security.PrincipalType;
+import io.trino.testing.AbstractTestQueryFramework;
+import io.trino.testing.LocalQueryRunner;
+import org.testng.annotations.AfterClass;
+import org.testng.annotations.Test;
+
+import java.io.File;
+import java.io.IOException;
+import java.io.UncheckedIOException;
+import java.nio.file.Files;
+import java.util.Optional;
+
+import static com.google.common.io.MoreFiles.deleteRecursively;
+import static com.google.common.io.RecursiveDeleteOption.ALLOW_INSECURE;
+import static com.google.inject.util.Modules.EMPTY_MODULE;
+import static io.trino.testing.TestingSession.testSessionBuilder;
+import static java.lang.String.format;
+
+@Test(singleThreaded = true)
+public class TestIcebergHiveMetastoreTableOperationsReleaseLockFailure
+        extends AbstractTestQueryFramework
+{
+    private static final String ICEBERG_CATALOG = "iceberg";
+    private static final String SCHEMA_NAME = "test_schema";
+    private File baseDir;
+
+    @Override
+    protected LocalQueryRunner createQueryRunner()
+    {
+        Session session = testSessionBuilder()
+                .setCatalog(ICEBERG_CATALOG)
+                .setSchema(SCHEMA_NAME)
+                .build();
+
+        try {
+            baseDir = Files.createTempDirectory(null).toFile();
+        }
+        catch (IOException e) {
+            throw new UncheckedIOException(e);
+        }
+
+        LocalQueryRunner queryRunner = LocalQueryRunner.create(session);
+
+        ThriftMetastore thriftMetastore = createMetastoreWithReleaseLockFailure();
+        TestingIcebergHiveMetastoreCatalogModule testModule = new TestingIcebergHiveMetastoreCatalogModule(thriftMetastore);
+        HiveMetastore metastore = testModule.getHiveMetastore();
+
+        queryRunner.createCatalog(
+                ICEBERG_CATALOG,
+                new TestingIcebergConnectorFactory(Optional.of(testModule), Optional.empty(), EMPTY_MODULE),
+                ImmutableMap.of());
+
+        Database database = Database.builder()
+                .setDatabaseName(SCHEMA_NAME)
+                .setOwnerName(Optional.of("public"))
+                .setOwnerType(Optional.of(PrincipalType.ROLE))
+                .build();
+        metastore.createDatabase(database);
+
+        return queryRunner;
+    }
+
+    private InMemoryThriftMetastore createMetastoreWithReleaseLockFailure()
+    {
+        return new InMemoryThriftMetastore(new File(baseDir + "/metastore"), new ThriftMetastoreConfig()) {
+            @Override
+            public long acquireTableExclusiveLock(AcidTransactionOwner transactionOwner, String queryId, String dbName, String tableName)
+            {
+                // returning dummy lock
+                return 100;
+            }
+
+            @Override
+            public void releaseTableLock(long lockId)
+            {
+                throw new RuntimeException("Release table lock has failed!");
```

Comment:
`RuntimeException` will be thrown when metastore faield to release lock? It would be nice to throw the same exception class if you already know it.

Reply:
Underlying `ThriftHiveMetastoreClient` can throw `TException` which extends `Exception` but that we can't throw as the `releaseTableLock` doesn't throw by `ThriftMetastore`

Reply:
@ebyhr is this ok with you?